### PR TITLE
Rhessi srm editable (post rename)

### PR DESCRIPTION
**Pull request by settwi**
Addresses issue #116 by
- auto-picking the correct RHESSI SRM based on the selected time range of the data.
- making it easy to add systematic error e.g.: `sunkit_fitter.systematic_error = 0.1`

Also deletes the STIX loader. I think I did that in the past when I was frustrated with how it (didn't really) work.
We should probably add that back before this gets merged... if it ever does :p

This change makes it actually possible to do good analysis on RHESSI data.
I am going to use this branch in a paper so it would be great if we could get the changes merged in.

\- WS

### CODE REVIEW

**Comment by KriSun95** [comment]
This PR is looking good. Would it be worth while creating the `extern` module in `fitting_legacy` and moving the RHESSI code there in this PR as well?

**Comment by samaloney** [comment]
> This PR is looking good. Would it be worth while creating the `extern` module in `fitting_legacy` and moving the RHESSI code there in this PR as well?

The `extern` module is already there just need to move the the code and maybe convert the `ipynb` to `py` for easier diffs.

**Comment by settwi** [comment]
@DanRyanIrish @samaloney @KriSun95 

pre-comit.ci is failing because autopep8, etc. are failing on poorly-formatted code in lots of files. i don't really want to touch them in this commit.

**Comment by samaloney** [comment]
I think you'll either need to rebase or merge in main as theses have all been fixed and pre-commit passes on main.

**Comment by settwi** [comment]
ok. i'll work on it

**Comment by samaloney** [comment]
It looks like pre-commit was able to fix all the issues so if you run `pre-commit run -a` double check the diff and then add the changes should be good to go.

**Comment by settwi** [comment]
oh wow, i didn't know you could do that :p

**Comment by settwi** [comment]
ruh roh readthedox failed one second.

**Review by samaloney — COMMENTED** [review]
LGTM. If the RHESSI loader tests are still valid would be good to move them over to extern too.

**Comment by settwi** [comment]
I’m pretty sure they aren’t valid because i condensed the code. but we could add some version back if need be.

**Comment by samaloney** [comment]
I think it would be good but as this is't in the main package I'm not going to hold it up over that but I would strongly encourage you too add some tests at some point 😉

**Review by samaloney — APPROVED** [review]

**Review by KriSun95 — APPROVED** [review]

File: sunkit_spex/fitting_legacy/instruments.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ class InstrumentBlueprint:
     as **kwargs.
 
     The `DataLoader` class in data_loader.py then creates a dictionary attrbute called `loaded_spec_data`
-    (note no underscore) that is then getable by the user when spectral fitting with the `Fitter` class
+    (note no underscore) that is then getable by the user when spectral fitting with the `SunXspex` class
```
[the remark this answers is not in the document]
Comment (DanRyanIrish) [reply sunkit_spex/fitting_legacy/instruments.py:52]:
```suggestion
    (note no underscore) that is then getable by the user when spectral fitting with the `Fitter` class
```

**Pull request merged**